Documentation - Project Settings Maya (#57)
ynput/ayon-documentation: website/docs/project_settings/settings_project_maya.md

---
id: settings_project_maya
title: Project Maya Settings
sidebar_label: Maya
---

# Open Workfile Post Initialization

> Open the Workfile on Maya's launch.

Reviewer: ```suggestion
Open the Workfile on Maya's launch. This changes default behaviour of opening last Workfile (that works by passing Workfile as argument when launching Maya). When enabled, last Workfile is loaded after Maya is fully initialized. This can help in certain scenarios where Workfile depends on Maya plugins that are not fully initialized yet when the scene is already opened.
```

# Explicit Plugins Loading
Determine the plugins to load on Maya's launch.

# Color Management (OCIO managed)
//

# Maya MEL Workspace
Text to set in the MEL Workspace file.

# Extension Mapping
Save a family (on left) as the indicated extension (on right). 

![Extension Mapping](assets/settings_project_maya_extension_mapping.png)

# Maya Directory Mapping
1. Enable it to use the environment variables in the referenced paths.
2. Put here your Path
3. Put here your desination path

The **source path** will be replaced by the **destination path**. 

![Maya Directory Mapping](assets/settings_project_maya_MayaDirectoryMapping.png)

# Include/Exclude Handles in default playback & render range
1. Enable or disable the inclusion of handles by default.
2. You can also define the inclusion/exclusion of handles by task type.

![Include/Exclude Handles](assets/settings_project_maya_extension_handles.png)

# Scripts Menu Definition
Build a menu in Maya.
![Script Menu Definition](assets/settings_project_maya_ScriptMenuDefinition.png)

> For exemple here : 

Reviewer: Typo
```suggestion
For example here: 
```

- **Menu Name :** Write your menu name
- **Menu Definition :**

→ "type": define the button type. Here, it's an action because when you click on it, it directly loads another window or commit any other action you defined in your script. 
*(if you define the type as a "menu", you can unfold the button to see other buttons and define them as actions).*

![Menu](assets/settings_project_maya_PublishGUIFilters_Menu.png)

→ "command": define the command to implement the script in the button.

> → "sourcetype": define the programmation langage of the script.

Reviewer: ```suggestion
→ "sourcetype": define the programming language of the script.
```
